### PR TITLE
Manage yumrepos via data

### PR DESCRIPTION
**Pull request by lamawithonel**
This commit adds two parameters to manage Yumrepo resources, as well as some default data.  The first parameter, `repos`, is simply a hash of Yumrepo resources.  The second parameter, `managed_repos`, takes a list of names-- keys to the `repos` hash-- and uses them to select data from the `repos` parameter hash.  The selected Yumrepos are then realized via an each loop, similar to an old `create_resources()` call.  The data includes most of the default Yumrepos for CentOS 6 & 7, with the exception of the "archive" repos.

### CODE REVIEW

**Review by cmentzer — APPROVED** [review]
Cursory review looks fine. Consider changing style in hiera.yaml in the future?

**Review by jeefberkey — COMMENTED** [review]
I know there is a pretty solid example of the data structure in the CentOS.yaml, but a mention in the README wouldn't hurt.

**Review by trevor-vaughan — COMMENTED** [review]
Looks OK. @jeefberkey Did you get a chance to test this live?

**Comment by lamawithonel** [comment]
I've submitted this [upstream](https://github.com/voxpupuli/puppet-yum/pull/40). Please make any further comments there.

**Comment by jeefberkey** [comment]
@trevor-vaughan not yet, but this would be a new addition, unused by default. I'm OK with merging this.

**Review by nick-markowski — APPROVED** [review]
I have no problem putting this out there, detached from simp-core.

File: functions/bool2num_hash_recursive.pp [hunk 1]
```diff
@@ -0,0 +1,13 @@
+function yum::bool2num_hash_recursive($arg) {
```
Comment (jeefberkey) [inline functions/bool2num_hash_recursive.pp:33]: This function could use some docs

Comment (lamawithonel) [reply functions/bool2num_hash_recursive.pp:33]: fin ack

**Comment by jeefberkey** [comment]
This was merged upstream.

**Comment by lamawithonel** [comment]
This was merged upstream. See voxpupuli#40.

**Pull request closed without merging**